tree/treeplayer/src/TTreeReaderValue.cxx

```diff
@@ -158,7 +216,9 @@ std::string ROOT::Internal::TTreeReaderValueBase::GetElementTypeName(const std::
 /// The TTreeReader has switched to a new TTree. Update the leaf.
 
 void ROOT::Internal::TTreeReaderValueBase::NotifyNewTree(TTree* newTree) {
-   if (!fHaveLeaf)
+   fProxyReadFunc = &TTreeReaderValueBase::ProxyReadDefaultImpl;
```

Comment:
Could you add a comment on why switching the tree requires resetting the read function? That's very much not obvious to me.

Reply:
Will add a comment along the line of 'new TTree might have a different structure and features ...'